mbed-client-mbedtls/m2mconnectionsecuritypimpl.h

```diff
@@ -21,32 +21,25 @@
 #include "mbed-client/m2mtimerobserver.h"
 #include "mbed-client/m2mconstants.h"
 #include "mbed-client/m2msecurity.h"
-
-#include "mbedtls/config.h"
-#include "mbedtls/platform.h"
-#include "mbedtls/debug.h"
-#include "mbedtls/ssl.h"
-#include "mbedtls/entropy.h"
-#include "mbedtls/ctr_drbg.h"
-#include "mbedtls/error.h"
-#include "mbedtls/certs.h"
-#include "mbedtls/entropy_poll.h"
+extern "C"{
+#include "pal_TLS.h"
+}
 
 class M2MTimer;
 
 //TODO: Should we let application to select these or not??
```

Comment:
If we can comment this out and it doesnt' matter, should we not just remove the whole block?